routes/gamesRouter.js

```diff
@@ -1,163 +1,176 @@
-    const gambling = req.body.gambling;
-    const explicitLanguage = req.body.explicitLanguage;
-    const averageScore = req.body.averageScore;
-    const publisher = req.body.publisher;
-    const releaseDate = req.body.releaseDate;
-    const query = {};
-    if (title) {
-        query.title = title;
-    }
-    if (category) {
-        query.category = category;
-    }
-    if (ageRating) {
-        query.ageRating = ageRating;
-    }
-    if (violence) {
-        query.violence = violence;
-    }
-    if (sexAndNudity) {
-        query.sexAndNudity = sexAndNudity;
-    }
-    if (alcoholAndDrugs) {
-        query.alcoholAndDrugs = alcoholAndDrugs;
-    }
-    if (gambling) {
-        query.gambling = gambling;
-    }
-    if (explicitLanguage) {
-        query.explicitLanguage = explicitLanguage;
-    }
-    if (averageScore) {
-        query.averageScore = averageScore;
-    }
-    if (publisher) {
-        query.publisher = publisher;
-    }
-    if (releaseDate) {
-        query.releaseDate = releaseDate;
-    }
+router.post("/search", (req, res) => {
+  const {
+    title,
+    category,
+    ageRating,
+    violence,
+    sexAndNudity,
+    alcoholAndDrugs,
+    gambling,
+    explicitLanguage,
+    averageScore,
+    publisher,
+    releaseDate,
+  } = req.body;
+  const query = {};
+  if (title) {
+    query.title = title;
+  }
+  if (category) {
+    query.category = category;
+  }
+  if (ageRating) {
+    query.ageRating = ageRating;
+  }
+  if (violence) {
+    query.violence = violence;
+  }
+  if (sexAndNudity) {
+    query.sexAndNudity = sexAndNudity;
+  }
+  if (alcoholAndDrugs) {
+    query.alcoholAndDrugs = alcoholAndDrugs;
+  }
+  if (gambling) {
+    query.gambling = gambling;
+  }
+  if (explicitLanguage) {
+    query.explicitLanguage = explicitLanguage;
+  }
+  if (averageScore) {
+    query.averageScore = averageScore;
+  }
+  if (publisher) {
+    query.publisher = publisher;
+  }
+  if (releaseDate) {
+    query.releaseDate = releaseDate;
+  }
 
-    Game.find(query, (err, results) => {
-        if (err) {
-            console.log(err);
-            res.status(204).json([]);
-        } else {
-        res.status(200).json(results);
-        }
-    });
+  Game.find(query, (err, results) => {
+    if (err) {
+      console.log(err);
+      res.status(204).json([]);
+    } else {
+      res.status(200).json(results);
+    }
+  });
 });
 
-router.get('/id/:id', (req, res) => {
-    Game.findOne({_id: req.params.id}, (err, game) => {
-        if (err) {
-            console.log(err);
-            res.status(404).json({err: `The game ${req.params.id} does not exist!`});  
-        } else {
-            res.status(200).json(game);
-        }
-    });
+router.get("/id/:id", (req, res) => {
+  Game.findOne({ _id: req.params.id }, (err, game) => {
+    if (err) {
+      console.log(err);
+      res
+        .status(404)
+        .json({ err: `The game ${req.params.id} does not exist!` });
```

Comment:
Remember, there may be more errors than the game just not existing.

Reply:
I think Maddy handled this stuff on Friday, I'm leaving it until she merges her stuff then can come back to it.